extensions/aris/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -0,0 +1,493 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as nls from 'vscode-nls';
+import * as azdata from 'azdata';
+import { IControllerError, IEndPointsResponse } from '../controller/types';
+import { ArisController } from '../controller/controller';
+
+const localize = nls.loadMessageBundle();
+
+export class AddControllerDialog {
+
+	private readonly DialogTitle: string = localize('bigDataClusters.addControllerDialog.title', 'Add New Controller');
+	private readonly SignInButtonText: string = localize('bigDataClusters.addControllerDialog.ok.title', 'Add');
+	private readonly CancelButtonText: string = localize('bigDataClusters.addControllerDialog.cancel.title', 'Cancel');
+
+	private dialog: azdata.window.Dialog;
+	private modelBuilder: azdata.ModelBuilder;
+
+	private urlInputBox: azdata.InputBoxComponent;
+	private usernameInputBox: azdata.InputBoxComponent;
+	private passwordInputBox: azdata.InputBoxComponent;
+	private rememberPwCheckBox: azdata.CheckBoxComponent;
+	private errorTextContainer: azdata.FlexContainer;
+	private callback: (e: IEndPointsResponse, rememberPassword: boolean) => any;
+	private callbackCancel: () => any;
+	private callbackError: (e: IControllerError) => any;
+
+	private prefilledValues: {
```

Comment:
I synced with Gene and recommended using a AddControllerModel class that has the values and methods for complete, canceled and error instead of the complicated mix of stuff here. I believe he will do that

Reply:
fixed.